.github/workflows/install-valkey/action.yml

```diff
@@ -0,0 +1,70 @@
+name: Install Valkey
+
+inputs:
+    valkey-version:
+        description: "valkey version to install"
+        required: true
+        type: string
+    target:
+        description: "Specified target toolchain, ex. x86_64-unknown-linux-gnu"
+        type: string
+        required: true
+        options:
+            - x86_64-unknown-linux-gnu
+            - aarch64-unknown-linux-gnu
+            - x86_64-apple-darwin
+            - aarch64-apple-darwin
+            - aarch64-unknown-linux-musl
+            - x86_64-unknown-linux-musl
+
+env:
+    CARGO_TERM_COLOR: always
+
+runs:
+    using: "composite"
+
+    steps:
+        - name: Cache Valkey
+          uses: actions/cache@v4
+          id: cache-valkey
+          with:
+            path: |
+                ~/valkey
+            key: valkey-${{ inputs.valkey-version }}-${{ inputs.target }}
+
+        - name: Build Valkey
+          if: ${{ steps.cache-valkey.outputs.cache-hit != 'true' }}
+          shell: bash
+          run: | 
+            echo "Building valkey ${{ inputs.valkey-version }}"
+            cd ~
+            git clone https://github.com/valkey-io/valkey.git
+            cd valkey
+            git checkout ${{ inputs.valkey-version }}
+            make BUILD_TLS=yes
+
+        - name: Install Valkey
+          shell: bash
+          run: |
+            cd ~/valkey
+            sudo make install
+            echo 'export PATH=/usr/local/bin:$PATH' >>~/.bash_profile
```

Comment:
why do we need this? AFAIK `make install` copies binaries to `/usr/bin`, so they become available in `$PATH`